schemas/process.yml

```diff
@@ -197,3 +235,94 @@
+      short: Whether the process is connected to an interactive shell.
+      description: >
+        Whether the process is connected to an interactive shell.
+        Process interactivity is inferred from the processes file descriptors. If the character device for the controlling tty is the same as stdin and stderr for the process, the process is considered interactive.
+        Note: A non-interactive process can belong to an interactive session and is simply one that does not have open file descriptors reading the controlling TTY on FD 0 (stdin) or writing to the controlling TTY on FD 2 (stderr). A backgrounded process is still considered interactive if stdin and stderr are connected to the controlling TTY.
+   
+    - name: same_as_process
+      level: extended
+      type: boolean
+      example: true
+      beta: This field is beta and subject to change.
+      short: This boolean is used to identify if a leader process is the same as the top level process.
+      description: >
+        This boolean is used to identify if a leader process is the same as the top level process.
+        For example, if process.group_leader.same_as_process = true, it means the process event in question is the leader of its process group. Details under process.* like pid would be the same under process.group_leader.*
+        The same applies for both process.session_leader and process.entry_leader.
+        This field exists to the benefit of EQL and other rule engines since it's not possible to compare equality between two fields in a single document.
+        e.g
+        process.entity_id = process.group_leader.entity_id (top level process is the process group leader)
+        OR
+        process.entity_id = process.entry_leader.entity_id (top level process is the entry session leader)
+        Instead these rules could be written like:
+        process.group_leader.same_as_process: true
+        OR
+        process.entry_leader.same_as_process: true
+        Note: This field is only set on process.entry_leader, process.session_leader and process.group_leader.
+
+    - name: env_vars
+      level: extended
+      type: object
+      beta: This field is beta and subject to change.
+      short: Environment variables set at the time of the event.
+      description: >
+        Environment variables set at the time of the event.
+        May be filtered to protect sensitive information.
+      example: >
+        {
+          "USER": "elastic",
+          "LANG": "en_US.UTF-8",
+          "HOME": "/home/elastic"
+        }
+
+    - name: entry_meta.type
+      level: extended
+      type: keyword
+      beta: This field is beta and subject to change.
+      short: The entry type for the entry session leader.
+      description: >
+        The entry type for the entry session leader.
+        Values include: init(e.g systemd), sshd, ssm, kubelet, teleport, terminal, console
+
+    - name: entry_meta.source
```

Comment:
Just realized we have this defined here as well as in source.yml. I would guess we should omit from process.yml?

Reply:
do we need a -name: entry_meta to replace it? so source.yml has something to attach to?  kinda like below with the 'tty' field?

Reply:
Good catch, I will remove it from process

We can define it if you want to be explicit, but it isn't necessary